waspc/src/Generator/ServerGenerator/QueryGenerator.hs

```diff
@@ -0,0 +1,84 @@
+module Generator.ServerGenerator.QueryGenerator
+    ( genQueries
+    , queryRouteInQueriesRouter
+    ) where
+
+import Data.Maybe (fromJust)
+import Data.Aeson ((.=), object)
+import qualified Path as P
+
+import qualified Util as U
+import StrongPath (Path, Rel, File, Dir, (</>))
+import qualified StrongPath as SP
+import Wasp (Wasp)
+import qualified Wasp
+import qualified Wasp.Query
+import qualified Wasp.JsImport
+import Generator.FileDraft (FileDraft)
+import qualified Generator.ServerGenerator.Common as C
+
+genQueries :: Wasp -> [FileDraft]
+genQueries = genQueryRoutes
+
+genQueryRoutes :: Wasp -> [FileDraft]
+genQueryRoutes wasp = concat
+    [ map (genQueryRoute wasp) (Wasp.getQueries wasp)
+    , [genQueriesRouter wasp]
+    ]
+
+genQueryRoute :: Wasp -> Wasp.Query.Query -> FileDraft
+genQueryRoute _ query = C.makeTemplateFD tmplFile dstFile (Just tmplData)
+  where
+    tmplFile = C.asTmplFile [P.relfile|src/routes/queries/_query.js|]
+
+    dstFile = queryRoutesDirInServerRootDir </> queryRouteFileInQueryRoutesDir query
+
+    tmplData = object
+        [ "queryJsFnImportStatement" .= ("import " ++ importList ++ " from '" ++ fromPath ++ "'")
+        , "queryJsFnIdentifier" .= importIdentifier
+        ]
+
+    jsQueryImport = Wasp.Query._jsFunction query
+
+    (importIdentifier, importList) =
```

Comment:
you have here `importList` but it is actually not a list (sometimes not even conceptually, when there is only default import), so maybe that is a bit confusing. I called that `importWhat` I think. Maybe `whatToImport` would be even better.

Reply:
True, I thought `importWhat` is not a very nice name so I went with importList, since I thought ok, this part of import statement is a list of possible imports. But I renamed it back to importWhat, it is probably clearer.